static-assets/components/cstudio-search/search.js

```diff
@@ -792,6 +835,85 @@
         CStudioAuthoring.Service.lookupContentItem(CStudioAuthoringContext.site, path, callback, false, false);
     }
 
+    CStudioSearch.saveContent = function() {
+        var searchId = this.searchContext ? this.searchContext.searchId : "" ;
+        var crossServerAccess = false;
+        var opener = window.opener ? window.opener : parent.iframeOpener;
+
+        try {
+            // unfortunately we cannot signal a form close across servers
+            // our preview is in one server
+            // our authoring is in another
+            // in this case we just close the window, no way to pass back details which is ok in some cases
+            if(opener.CStudioAuthoring) { }
+        }
+        catch(crossServerAccessErr) {
+            crossServerAccess = true;
+        }
```

Comment:
You may do this through window.postMessage. Environment values supplied by `CStudioAuthoringContext` provide the host origin and other necessary info required to communicate between windows.

This is how host/guest work and there are also examples of this sort of communication in forms-engine.

Ping me if you'd like to discuss further or get more details.